### ISSUE TITLE
Move important information from Wiki to Docs

### ISSUE THREAD

**Issue by wtgodbe** [dotnet/corefx#36057]
We should go through https://github.com/dotnet/corefx/wiki & move anything important to https://github.com/dotnet/corefx/tree/master/Documentation, then abandon the wiki.

CC @karelz @danmosemsft

**Comment by davidsh** [comment]
>We should go through https://github.com/dotnet/corefx/wiki & move anything important to https://github.com/dotnet/corefx/tree/master/Documentation, then abandon the wiki.

I think some of the information needs to be placed on doc.microsoft.com in the appropriate place. For example, some behavior differences between .NET Framework and .NET Core for networking APIs should be documented in the *Remarks* section of the appropriate API.

E.g.: [HttpResponseMessage.EnsureSuccessStatusCode](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpresponsemessage.ensuresuccessstatuscode?view=netcore-2.2#remarks)

**Comment by ViktorHofer** [comment]
This needs to happen before we merge repositories. cc @carlossanlop @Anipik in case you are interested helping here.

**Comment by ViktorHofer** [comment]
@carlossanlop would you want to help out here? We need to do this before the repo consolidation happens.

**Comment by carlossanlop** [comment]
Sure I can help on this.
cc @karelz

**Comment by karelz** [comment]
Yesss! I hope we can retire the XXXL long docs in Docs directory in favor of the new-contributor- friendly style docs we have on wiki 🤗
De-duplication would be first great step.

**Comment by ViktorHofer** [comment]
Great, Carlos I'll send you some ideas.

**Comment by carlossanlop** [comment]
@ViktorHofer @karelz here is my plan, 5 stages:

- [ ] Backup all the `*.md` files located in the wiki repo [dotnet/corefx.wiki](https://github.com/dotnet/corefx.wiki) inside a **temporary** folder in the [dotnet/corefx](https://github.com/dotnet/corefx) repo, so we don't block the repo merge work, and we don't worry about deleting that info permanently. I will do this immediately.

- [ ] Keep the relevant `*.md` files, delete the useless/obsolete ones. I went through all the files and this is my initial proposal (pending verify the "Maybe" files):

|Keep|*.md filename|Why|
|-|-|-|
|<span style="color:red">No</span>|[_Sidebar.md](https://github.com/dotnet/corefx/wiki/_Sidebar.md)|Nothing useful|
|<span style="color:green">Yes</span>|[ApiCompat.md](https://github.com/dotnet/corefx/wiki/ApiCompat.md)|Historical info|
|<span style="color:red">No</span>|[Build-and-run-tests.md](https://github.com/dotnet/corefx/wiki/Build-and-run-tests.md)|Already documented [here](https://github.com/dotnet/corefx/blob/017a038abaabedcbbc61c85c9f7405f588ca3b46/Documentation/project-docs/developer-guide.md) and [here](https://github.com/dotnet/corefx/blob/c22714d65151968a17c3b232748880d5e7961dee/Documentation/debugging/debugging-packages.md)|
|Maybe|[Building-.NET-Core--2.x-on-FreeBSD.md](https://github.com/dotnet/corefx/wiki/Building-.NET-Core--2.x-on-FreeBSD.md)|Do we need to preserve the 2.X instructions?|
|<span style="color:red">No</span>|[Building-.NET-Core-3.x-on-FreeBSD.md](https://github.com/dotnet/corefx/wiki/Building-.NET-Core-3.x-on-FreeBSD.md)|Already documented [here](https://github.com/dotnet/corefx/blob/2c5f652ea6a883bc032db18a8a2401db9f66f523/Documentation/building/unix-instructions.md)|
|<span style="color:red">No</span>|[Checking-out-the-code-repository.md](https://github.com/dotnet/corefx/wiki/Checking-out-the-code-repository.md)|We should not maintain a git tutorial|
|<span style="color:red">No</span>|[Contributing.md](https://github.com/dotnet/corefx/wiki/Contributing.md)|Nothing useful|
|<span style="color:red">No</span>|[Contributions.md](https://github.com/dotnet/corefx/wiki/Contributions.md)|Nothing relevant today|
|<span style="color:red">No</span>|[Creating-a-Pull-Request.md](https://github.com/dotnet/corefx/wiki/Creating-a-Pull-Request.md)|We should not maintain a git tutorial. PR rules are documented [here](https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/Documentation/project-docs/contributing.md).|
|<span style="color:red">No</span>|[Development-environment-setup-for-Linux.md](https://github.com/dotnet/corefx/wiki/Development-environment-setup-for-Linux.md)|Already in docs, and up to date|
|<span style="color:red">No</span>|[Development-environment-setup-for-MacOS.md](https://github.com/dotnet/corefx/wiki/Development-environment-setup-for-MacOS.md)|Already in docs, and up to date|
|<span style="color:red">No</span>|[Development-environment-setup-for-Windows.md](https://github.com/dotnet/corefx/wiki/Development-environment-setup-for-Windows.md)|Already in docs, and up to date|
|<span style="color:green">Yes</span>|[Enabling-New-OS.md](https://github.com/dotnet/corefx/wiki/Enabling-New-OS)|Useful instruction links|
|<span style="color:red">No</span>|[git-reference.md](https://github.com/dotnet/corefx/wiki/git-reference.md)|We should not maintain a git tutorial|
|<span style="color:red">No</span>|[Hackathon.md](https://github.com/dotnet/corefx/wiki/Hackathon.md)|Obsolete event|
|<span style="color:red">No</span>|[Home.md](https://github.com/dotnet/corefx/wiki/Home.md)|Nothing relevant|
|Maybe|HTTP-2-Implementaiton-Status.md|Ask networking|
|<span style="color:red">No</span>|[Issues.md](https://github.com/dotnet/corefx/wiki/Issues.md)|Nothing useful|
|<span style="color:green">Yes</span>|[Joining-.NET-Core-team.md](https://github.com/dotnet/corefx/wiki/Joining-.NET-Core-team.md)|Save this|
|<span style="color:green">Yes</span>|[Links.md](https://github.com/dotnet/corefx/wiki/Links.md)|Keep the conference links, remove irrelevant links|
|<span style="color:red">No</span>|[New-contributor-docs---TODO-list.md](https://github.com/dotnet/corefx/wiki/New-contributor-docs---TODO-list.md)|Nothing useful|
|<span style="color:green">Yes</span>|[New-contributor-docs.md](https://github.com/dotnet/corefx/wiki/New-contributor-docs.md)|Preserve only some of the hyperlinks|
|<span style="color:green">Yes</span>|[Pick-Issue.md](https://github.com/dotnet/corefx/wiki/Pick-Issue.md)|Summarize the label descriptions|
|<span style="color:red">No</span>|[Setting-up-the-development-environment.md](https://github.com/dotnet/corefx/wiki/Setting-up-the-development-environment.md)|Nothing useful|
|<span style="color:green">Yes</span>|[UWP-Compat.md](https://github.com/dotnet/corefx/wiki/UWP-Compat.md)|Historical info|

- [ ] Move the information in the wiki `*.md` files into their relevant section in the corefx Documentation folder.

- [ ] Restructure the documentation. @ViktorHofer offered a suggested initial structure, we can discuss/expand it further:

	- docs
		- design
		- project
		- manpages
		- bot
		- coding-guidelines
		- workflow
			- building
			- testing
			- profiling
				- performance

- [ ] Remove duplication. I found many topics that are explained in multiple places, we should substitute them all with a link to a central, unique `*.md` page. For example, searching for `RebuildAndTest` instructions in the corefx repo [returns many pages where running unit tests is explained](https://github.com/search?q=repo%3Adotnet%2Fcorefx+extension%3Amd+path%3A%2FDocumentation+rebuildandtest&type=Code).

**Comment by carlossanlop** [comment]
Update: I was told the corefx repo and the wiki won't get destroyed, so I'll skip step 1 from my previous comment.
I will focus on only bringing the relevant information into our repo.

**Comment by jkotas** [comment]
> ApiCompat.md	Historical info

This information maintained in documentation and https://github.com/Microsoft/dotnet-apiport database. We do not need a stale copy in the repo.

> HTTP-2-Implementaiton-Status.md	Networking team is actively using it

github issues and projects are the appropriate tools for tracking project status. This should not be commited as file. Convert to issue or project.

> Joining-.NET-Core-team.md	Save this

Information specific to Microsoft employees should not live on github. It should live in an internal OneNote.

**Comment by danmosemsft** [comment]
> Information specific to Microsoft employees should not live on github. It should live in an internal OneNote.

I noticed it's already here:

https://github.com/dotnet/core/blob/master/Documentation/microsoft-team.md#configure-your-github-account-as-a-microsoft-employee-recommended

**Comment by ViktorHofer** [comment]
@carlossanlop can you please come up with a plan for the categorization? Would like to know which documents would go into which directory. Having a common set of categories with coreclr and core-setup will help us consolidate the docs at a later point.

**Comment by ViktorHofer** [comment]
As stated in the PR, we still want to disable the WIKI feature in Corefx. That's listed as pre-requisite for the consolidation as it's a good metric when docs are ready.

**Comment by jkotas** [comment]
> we still want to disable the WIKI feature in Corefx

That is nice to have. The Wiki was limping along in CoreFX for number of years. It is not a big deal if it continues limping along in the archived CoreFX forever.

**Comment by carlossanlop** [comment]
@jkotas thanks for sharing your findings. Based on what you shared:
- I will not add the ApiCompat document since it's stale.
- I will ask the Networking team to delete this document and convert it to individual issues. We won't backup that file, so they should do it soon.
- Regarding the OneNote with Microsoft onboarding information, I have conflicting information because @danmosemsft shared a link [to a page](https://github.com/dotnet/core/blob/master/Documentation/microsoft-team.md#configure-your-github-account-as-a-microsoft-employee-recommended) containing instructions for MS employee onboarding. It seems we don't have a problem sharing this kind of information publicly (and only MS employees can follow these instructions anyway). Based on this, I will not backup that wiki file containing information, but instead, I will add a hyperlink in our documentation pointing [to the page](https://github.com/dotnet/core/blob/master/Documentation/microsoft-team.md#configure-your-github-account-as-a-microsoft-employee-recommended) shared by Dan. That way we avoid duplication of information. Does this sound good or would you prefer something else?

@jkotas @ViktorHofer I noticed the majority of the documents we have in the corefx wiki contained documentation *planning* information, not the documentation itself. The original idea was to delete the wiki. At the very least, we should disable the one in corefx, and avoid creating one in the new repo. Otherwise, we will start duplicating documentation again. What do you think?

**Comment by danmosemsft** [comment]
> That is nice to have. The Wiki was limping along in CoreFX for number of years. It is not a big deal if it continues limping along in the archived CoreFX forever.

I'd rather delete it - since it causes confusion through having out of date info. I haven't gone through what's in there and it's good @carlossanlop is looking through it but I'm guessing it would not be a big loss to simply delete it.

**Comment by jkotas** [comment]
>  only MS employees can follow these instructions anyway

These piecemeal instructions are incomplete. The full instructions to get a new hire in CoreCLR/CoreFX running are on internal one note that we would not be comfortable sharing.

Internal instructions do not belong to open source repositories. It does not look good for external contributors, and these instructions are a risk in that somebody will try to update them with details that are not supposed to be public.

**Comment by jkotas** [comment]
> I noticed the majority of the documents we have in the corefx wiki contained documentation planning information, not the documentation itselt

These are all @karelz idea that he did not have time to finish.

**Comment by jkotas** [comment]
> avoid creating one in the new repo.

Yes, we should not even think about recreating the wiki in the new runtime repo.

**Comment by carlossanlop** [comment]
@ViktorHofer @danmosemsft @jkotas I modified the Documentation README.md to merge corefx and coreclr documentation, and include the wiki data I considered useful from my table above.
Can you please review it?

This is how it would look:
https://github.com/carlossanlop/corefx/tree/NewReadme/Documentation

I should not submit a PR yet because I first need to organize the files.

**Comment by jkotas** [comment]
> merge corefx and coreclr documentation

I do not think this issue is about merging corefx and coreclr documentation. Merging corefx and coreclr documentation will be a very large undertaking and I think it is better to be done only after everything is in the runtime repository. 

This issue is about what is at the top - nothing more, nothing less:

> go through https://github.com/dotnet/corefx/wiki & move anything important to https://github.com/dotnet/corefx/tree/master/Documentation, then abandon the wiki.

**Comment by carlossanlop** [comment]
@jkotas The documentation merge task will eventually have to be done, so I'll keep the branch in my fork with the proposal for future reference.

Regarding porting wiki info, I will submit a PR soon. The change will be small.

**Comment by ViktorHofer** [comment]
@carlossanlop can we disable the wiki now?

**Comment by carlossanlop** [comment]
@ViktorHofer Yes we can.

**Comment by ViktorHofer** [comment]
Thanks, I just disabled the wiki. I also cloned it locally - in case somebody wants to restore something, ping me.

**Comment by ViktorHofer** [comment]
Here's the link to a backup: https://microsofteur-my.sharepoint.com/:u:/g/personal/vihofer_microsoft_com/EfekSugu7JJHrC5_QwEbL_QBvgBs21tAb4GUsVhRH6Zf-A?e=2jzaCd
